src/core/thread.d

```diff
@@ -631,7 +633,8 @@ class Thread
         }
 
         // Start thread as non-suspendable, undone in thread_entryPoint
-        atomicStore(m_isInCriticalRegion, true);
+        synchronized (Thread.criticalRegionLock)
+            m_isInCriticalRegion = true;
```

Comment:
> Only that we're never waiting for the loader lock, do we?

The loader lock is handled by the OS. We don't deal with it explicitely, but the OS thread startup code does when it calls DllMain(DLL_THREAD_ATTACH) for all loaded DLLs before entering the thread entry point.